multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/UploadAppStep.java

```diff
@@ -63,17 +64,21 @@ public StepPhase executeAsyncStep(ProcessContext context) throws FileStorageExce
 
         MtaArchiveElements mtaArchiveElements = context.getVariable(Variables.MTA_ARCHIVE_ELEMENTS);
         String moduleFileName = mtaArchiveElements.getModuleFileName(applicationToProcess.getModuleName());
+        CloudControllerClient client = context.getControllerClient();
         if (moduleFileName == null) {
             getStepLogger().debug(Messages.NO_CONTENT_TO_UPLOAD);
+            if (applicationToProcess.getDockerInfo() != null) {
```

Comment:
If I understand correctly, this flow means that there isn't a file entry in the MTAR associated with this module because it is described as a docker application, is this right?
If so, how did we operate with docker apps so far?

Reply:
In v2 the docker apps do not require packages, they are just created as docker image apps and started. In v3 it is required to create a docker package to deploy the application. If there is no module name the application is probably a docker app but I added this check just in case if the user does not provide docker parameters otherwise a null pointer will occur.